### PR TITLE
[RISC-V] Allow enabling CFG_TEE_CORE_DEBUG

### PR DESCRIPTION
**Pull request by lyctw**
This series enables CFG_TEE_CORE_DEBUG, a crucial feature for early error detection on RISC-V platforms, such as assertions.

### CODE REVIEW

File: core/arch/riscv/mm/core_mmu_arch.c [hunk 1]
```diff
@@ -466,6 +466,7 @@ bool arch_va2pa_helper(void *va, paddr_t *pa)
 	int level = 0;
 	unsigned int idx = 0;
 	struct mmu_partition *prtn = core_mmu_get_prtn();
+	vaddr_t offset_mask = 0; /* page offset mask */
```
Comment (etienne-lms) [inline core/arch/riscv/mm/core_mmu_arch.c:469]: nitpicking: I don't think the inline comment is really useful.

File: core/arch/riscv/mm/core_mmu_arch.c [hunk 2]
```diff
@@ -628,16 +628,16 @@ static void set_user_va_idx(struct mmu_partition *prtn)
 			return;
 		}
 	}
-	if (user_va_idx < 0)
-		panic();
+
+	assert(core_mmu_user_va_range_is_defined());
```
Comment (etienne-lms) [inline core/arch/riscv/mm/core_mmu_arch.c:632]: This replaces a always embedded test with a debug config assertion. Looks fine but I think it should be stated in the commit message.

File: core/mm/core_mmu.c [hunk 3]
```diff
@@ -2381,7 +2381,14 @@ paddr_t virt_to_phys(void *va)
...
 
-#if defined(CFG_TEE_CORE_DEBUG)
+/*
+ * Don't use check_va_matches_pa() for RISC-V, as its callee
+ * arch_va2pa_helper() will call it eventually, this creates
+ * indirect recursion and can lead to a stack overflow.
+ * Moreover, if arch_va2pa_helper() returns true, it implies
+ * the va2pa mapping is matched, no need to check again.
+ */
+#if defined(CFG_TEE_CORE_DEBUG) && !defined(RV64)
```
Comment (etienne-lms) [inline core/mm/core_mmu.c:2391]: Only for `RV64`, not for `RV32`? it seems the implementation of the same for these 2 configurations.

**Review by etienne-lms — COMMENTED** [review]
For commit "core: riscv: core_mmu_arch: fix arch_va2pa_helper() on superpage translation": commit message header line is a bit too long: could you shorten, e.g.:
"core: riscv: mm: fix arch_va2pa_helper() on superpage translation"

File: core/arch/riscv/mm/core_mmu_arch.c (hunk 2 above)
Comment (lyctw) [reply core/arch/riscv/mm/core_mmu_arch.c:632]: Upon reflection, I should keep the test unchanged.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the series.

**Comment by etienne-lms** [comment]
**CI Code style** error report seems due to a false positive.

**Comment by lyctw** [comment]
Tag applied, thanks!

**Pull request merged**